sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1307,86 +1310,70 @@ void CAFMaker::produce(art::Event& evt) noexcept {
         }
-        // Duplicate track reco info in the srslice
-        recslc.reco.trk.push_back(rec.reco.trk.back());
-        recslc.reco.ntrk = recslc.reco.trk.size();
       } // thisTrack exists
 
-      else if (!thisShower.empty()) { // it's a shower!
-        assert(thisTrack.size() == 0);
+      if (!thisShower.empty()) { // it has shower!
         assert(thisShower.size() == 1);
-        rec.reco.nshw ++;
-        rec.reco.shw.push_back(SRShower());
-        FillShowerVars(*thisShower[0], vertex, fmShowerHit.at(iPart), lar::providerFrom<geo::Geometry>(), producer, rec.reco.shw.back());
-
-        const larpandoraobj::PFParticleMetadata *pfpMeta = (iPart == fmPFPart.size()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
-        FillPFPVars(thisParticle, primary, pfpMeta, rec.reco.shw.back().pfp);
+        FillShowerVars(*thisShower[0], vertex, fmShowerHit.at(iPart), lar::providerFrom<geo::Geometry>(), producer, rec.reco.pfp.back().shw);
 
         // We may have many residuals per shower depending on how many showers ar in the slice
 
         if (fmShowerRazzle.isValid() && fmShowerRazzle.at(iPart).size()==1) {
-           FillShowerRazzle(fmShowerRazzle.at(iPart).front(), rec.reco.shw.back());
+           FillShowerRazzle(fmShowerRazzle.at(iPart).front(), rec.reco.pfp.back().shw);
         }
         if (fmShowerCosmicDist.isValid() && fmShowerCosmicDist.at(iPart).size() != 0) {
-          FillShowerCosmicDist(fmShowerCosmicDist.at(iPart), rec.reco.shw.back());
+          FillShowerCosmicDist(fmShowerCosmicDist.at(iPart), rec.reco.pfp.back().shw);
         }
         if (fmShowerResiduals.isValid() && fmShowerResiduals.at(iPart).size() != 0) {
-          FillShowerResiduals(fmShowerResiduals.at(iPart), rec.reco.shw.back());
+          FillShowerResiduals(fmShowerResiduals.at(iPart), rec.reco.pfp.back().shw);
         }
         if (fmShowerTrackFit.isValid() && fmShowerTrackFit.at(iPart).size()  == 1) {
-          FillShowerTrackFit(*fmShowerTrackFit.at(iPart).front(), rec.reco.shw.back());
+          FillShowerTrackFit(*fmShowerTrackFit.at(iPart).front(), rec.reco.pfp.back().shw);
         }
         if (fmShowerDensityFit.isValid() && fmShowerDensityFit.at(iPart).size() == 1) {
-          FillShowerDensityFit(*fmShowerDensityFit.at(iPart).front(), rec.reco.shw.back());
+          FillShowerDensityFit(*fmShowerDensityFit.at(iPart).front(), rec.reco.pfp.back().shw);
         }
         if (fmShowerHit.isValid()) {
-          if ( !isRealData ) FillShowerTruth(fmShowerHit.at(iPart), id_to_hit_energy_map, true_particles, clock_data, rec.reco.shw.back());
+          if ( !isRealData ) FillShowerTruth(fmShowerHit.at(iPart), id_to_hit_energy_map, true_particles, clock_data, rec.reco.pfp.back().shw);
         }
-        // Duplicate track reco info in the srslice
-        recslc.reco.shw.push_back(rec.reco.shw.back());
-        recslc.reco.nshw = recslc.reco.shw.size();
 
       } // thisShower exists
-
-      else {}
-
+      
```

Comment:
```suggestion
      
      rec.reco.pfp.push_back(std::move(pfp));
```
There, now I made it.